RefactoringsExperiments/src/main/java/org/jetbrains/research/extractMethodExperiments/extractors/CustomRefactoringHandler.java

```diff
@@ -66,45 +70,68 @@ private void handleCommit(List<Refactoring> refactorings) throws IOException {
-                if (filePath != null && locationInfo.getFilePath().equals(filePath.replace(cleanRepoPath, ""))) {
-                    PsiFile psiFile = PsiManager.getInstance(project).findFile(file);
-                    if (psiFile != null) {
-                        PsiMethod method = findMethodBySignature(psiFile, calculateSignature(extractedOperation));
-                        if (method != null) {
-                            writeFeaturesToFile(psiFile, method);
-                        }
-                    }
+            UMLOperation sourceOperation = extractOperationRefactoring.getSourceOperationBeforeExtraction();
+            LocationInfo sourceLocationInfo = sourceOperation.getLocationInfo();
+            CodeRange codeLocation = extractOperationRefactoring.getExtractedCodeRangeFromSourceOperation();
+            for (String path : changedSourceJavaFiles.keySet()) {
+                if (path.endsWith(sourceLocationInfo.getFilePath())) {
+                    PsiMethod dummyMethod = findMethodBySignature(changedSourceJavaFiles.get(path), calculateSignature(sourceOperation));
+                    String extractedCode = getMethodSlice(changedSourceJavaFiles.get(path),
+                            codeLocation.getStartLine(), codeLocation.getEndLine());
+                    writeFeaturesToFile(dummyMethod, sourceLocationInfo.getFilePath(),
+                            extractedCode, codeLocation.getStartLine(), codeLocation.getEndLine());
                 }
             }
         }
     }
 
-    private void writeFeaturesToFile(PsiFile psiFile, PsiMethod psiElement) throws IOException {
-        int beginLine = getNumberOfLine(psiFile, psiElement.getTextRange().getStartOffset());
-        int endLine = getNumberOfLine(psiFile, psiElement.getTextRange().getEndOffset());
-        MetricCalculator metricCalculator = new MetricCalculator(psiElement, beginLine, endLine);
+    private void writeFeaturesToFile(PsiMethod dummyPsiMethod, String filePath,
+                                     String code, int beginLine, int endLine) throws IOException {
+
+        Path tmpRepoPath = Paths.get(repositoryPath);
+        String repoName = tmpRepoPath.getName(tmpRepoPath.getNameCount()-1).toString();
+        MetricCalculator metricCalculator =
+                new MetricCalculator(code, dummyPsiMethod, repositoryPath, filePath, beginLine, endLine);
+
         FeaturesVector featuresVector = metricCalculator.getFeaturesVector();
 
         for (int i = 0; i < featuresVector.getDimension(); i++) {
             this.fileWriter.append(String.valueOf(featuresVector.getFeature(Feature.fromId(i))));
-            if (i != featuresVector.getDimension() - 1)
-                this.fileWriter.append(';');
+            this.fileWriter.append(';');
         }
 
+        this.fileWriter.append(repoName);
         this.fileWriter.append('\n');
     }
 
+    private static String getMethodSlice(PsiFile psiFile, int beginLine, int endLine) {
```

Comment:
why do we need this method? RefactoringMiner provides a method called `getExtractedCodeFragmentsFromSourceOperation()`.
why don't we use it?

Reply:
it returns some hashset with very strange objects in it, while i need some simple text